collagraph/collagraph.py

```diff
@@ -24,9 +25,31 @@
 VIRTUAL_NODE_TYPES = {"template", "slot"}
 
 
+class NotifyChangeWatcher(Watcher):
+    """
+    Custom watcher that simply calls callback function without evaluating
+    the new value of the watched expression.
+    This makes it possible to just signal to Collagraph that a dependency
+    has changed and then Collagraph can decide when to actually evaluate
+    the new value.
+    """
+
+    def update(self):
+        """On update, simply run the callback."""
+        self.callback()
```

Comment:
Is this really needed, when you can create lazy + sync watchers?

Reply:
Yes, this is really needed, because the important detail here is that the new value is _not_ evaluated before calling the callback. That is also why the callback is called immediately without any arguments (there is no 'new' value to pass to the callback).

Reply:
I see. I guess we could (in observ) create a base class `Watcher` that can only notify when the expression changes to support your use case here, and then subclass that with the `ValueWatcher` class as we know it.

In particular I think you are now (perhaps unintentionally) "working around" the observ scheduler, which protects against infinite update loops and redundant calls to the same watchers within a single event loop step. That part I think is really not something you want to miss out on.

Reply:
I see, that makes sense!

So what about `computed`? Do those still work?